### DISCUSSION TITLE
Edge Middleware Matching & Usage

### DISCUSSION THREAD

**Discussion by jamiemarkwhite** [vercel/community#6798] [Feature Question] [answered]
### Question

Hello, so I understand there are 2 primary documented ways to filter edge middleware calls: one being `export const config = {matcher:{...}}` and the other using `export default function middleware() {}` to include conditional logic. Would I be right in thinking the config matcher prevents the edge middleware from counting toward usage on filtered requests but everything that hits the default middleware version counts as a request, therefore counting toward usage regardless of a conditional match or not?

Thanks

**Comment by amyegan** [comment, marked as answer]
Yes, the [`config` option](https://vercel.com/docs/functions/edge-middleware/middleware-api#config-object) with `matcher` property is the preferred way to optimize [middleware usage](https://vercel.com/docs/functions/edge-middleware/usage-and-pricing#usage).

https://vercel.com/docs/pricing/edge-middleware#optimizing-middleware-invocations
